### PR TITLE
feat(vrf): era-specific VRF

### PR DESCRIPTION
**Pull request by wolf31o2**
---
## Summary by cubic
Make VRF inputs era-aware and verify KES against the original header body CBOR to fix leader selection and block validation across TPraos (Shelley–Alonzo) and CPraos (Babbage+).

- **New Features**
  - Era-specific VRF: TPraos uses MkSeedTPraos with SeedL; CPraos keeps MkInputVrf. Exposes SeedEta/SeedL.
  - KES verification uses the stored original body CBOR; added extractOriginalBodyCbor and unified ExtractKesFields across Shelley–Conway and Leios.

- **Bug Fixes**
  - Rolling nonce now follows the Cardano Nonce semigroup: blake2b256(a||b) with NeutralNonce as identity.
  - Stricter validation and clearer errors: require 32-byte eta0; reject slot overflows; error on unknown consensus mode; explicit VRF/KES failure messages.

<sup>Written for commit abf097809fa8809d35b4e6cdabbc355bfa238866. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Era-aware VRF/KES validation with clearer, component-specific error reporting and KES verification against original header CBOR; explicit errors for slot overflow and unknown consensus mode.

* **New Features**
  * Preservation of original header CBOR for more header types.
  * TPraos-specific VRF seed utilities exposed.

* **Behavioral Changes**
  * Rolling nonce now uses semigroup composition with identity handling and stricter input validation.

* **Tests**
  * Added TPraos seed tests, rolling-nonce validation & error tests, and leader-check edge-case tests.

### CODE REVIEW

File: ledger/verify_block.go [hunk 1]
```diff
@@ -119,6 +119,33 @@ func DetermineBlockType(headerCbor []byte) (uint, error) {
...
+		return h.Body.Cbor(), nil
+	case *mary.MaryBlockHeader:
+		return h.Body.Cbor(), nil
+	case *alonzo.AlonzoBlockHeader:
+		return h.Body.Cbor(), nil
+	case *babbage.BabbageBlockHeader:
+		return h.Body.Cbor(), nil
+	case *conway.ConwayBlockHeader:
+		return h.Body.Cbor(), nil
+	default:
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply ledger/verify_block.go]: Leios is experimental. This is fine for now.

**Review by agaffney — APPROVED** [review]
